src/coins.h

```diff
@@ -283,7 +283,20 @@ struct CAnchorsCacheEntry
         DIRTY = (1 << 0), // This cache entry is potentially different from the version in the parent view.
     };
 
-    CAnchorsCacheEntry() : entered(false), flags(0) {}
+    CAnchorsSproutCacheEntry() : entered(false), flags(0) {}
+};
+
+struct CAnchorsSaplingCacheEntry
```

Comment:
Is it possible to make this a template or something similar to avoid having to have the two classes?

Reply:
That's probably possible, though I think it should be explored in another PR.

Reply:
Sounds good to me.